kafka/producer/base.py

```diff
@@ -73,12 +79,16 @@ def _send_upstream(queue, client, codec, batch_time, batch_size,
                                  messages)
             reqs.append(req)
 
-        try:
-            client.send_produce_request(reqs,
-                                        acks=req_acks,
-                                        timeout=ack_timeout)
-        except Exception:
-            log.exception("Unable to send message")
+        for i in range(batch_send_max_retry):
+            try:
+                client.send_produce_request(reqs,
+                                            acks=req_acks,
+                                            timeout=ack_timeout)
+            except Exception:
+                log.exception("Unable to send message - retry {0}".format(i))
+                time.sleep(float(batch_send_retry_backoff_ms) / float(1000))
+                continue
+            break
```

Comment:
In the try or out the try do the same.
We continue if we face an exception or we break if no exception.